### ISSUE TITLE
Device Code flow fails with an error in the sign-in web app: "We don't recognize this code, please enter the code again"

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/azure-activedirectory-library-for-dotnet#1327]
**Which Version of ADAL are you using ?**
ADAL 4.2.0 RC

**Which platform has the issue?**
.NET Core

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [x ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

**Repro**
1. Run the following sample:  https://github.com/Azure-Samples/active-directory-dotnet-deviceprofile by following the README.md

**Expected behavior**
The sample works

**Actual behavior**
Whatever code is presented, when navigating to the URL, I get:
"We don't recognize this code, please enter the code again."
![image](https://user-images.githubusercontent.com/13203188/47011835-26df9100-d143-11e8-9e78-04c0ea184b9c.png)

**Comment by bgavrilMS** [comment]
This does not repro for me when using a sample app from the repo. This suggest that something is wrong in the configuration. I have seen this issue with a customer, in that case he mis configured the clientId. 

Will try with the sample and get back on this thread.

**Comment by jennyf19** [comment]
@jmprieur @bgavrilMS I haven't been able to reproduce this. I get the "we don't recognize this code" error when doing a copy/paste and including an extra space at the beginning or end. I also had to register the [sample app](https://github.com/Azure-Samples/active-directory-dotnet-deviceprofile) in my tenant (could not use the app out of the box), and now it's working.
Tested w/ADALv3.19.8 and current dev.
